internal/manifests/collector/route.go

```diff
@@ -86,5 +87,12 @@ func Routes(params manifests.Params) ([]*routev1.Route, error) {
 			},
 		}
 	}
-	return routes, nil
+
+	// Convert []*routev1.Route to []client.Object
+	result := make([]client.Object, len(routes))
+	for i, route := range routes {
+		result[i] = route
+	}
```

Comment:
Could this result in an improper copy because of shadowing?

Reply:
I'm not sure. Let me check and improve the code.